src/plugins/csp_configuration_provider/README.md

```diff
@@ -0,0 +1,64 @@
+# CspConfigurationProvider
+
+A OpenSearch Dashboards plugin
+
+This plugin introduces the support of self service dynamic configuration of Content Security Policy (CSP) rules without requiring a server restart. It registers a pre-response handler to `HttpServiceSetup` which can get CSP rules from a new index `.opensearch_dashboards_config` and then rewrite to CSP header. OSD users could update the content of the index with new CSP rules and the change will take effect immediately.
+
+It also provides an interface `CspClient` for future extensions of external CSP providers. By default, an implementation based on OpenSearch as database is used.
+
+---
+
+## Configuration
+
+By default, the new index does not exist. For OSD users who do not need customized CSP rules, there is no need to create this index. The system will then use whatever CSP rules aggregated by the values of `csp.rules` from `opensearch_dashboards.yml` and default values.
+
+Below is the current default value.
+
+```
+"script-src 'unsafe-eval' 'self'; worker-src blob: 'self'; style-src 'unsafe-inline' 'self'; frame-ancestors 'self'"
+```
+(Note that `frame-ancestors 'self'` is used to only allow self embedding and prevent clickjacking by default.)
+
+For OSD users who want to make changes, e.g allow this site `https://example-site.org` to embed OSD pages, they can update the index from DevTools Console.
+
+```
+PUT .opensearch_dashboards_config/_doc/csp.rules
```

Comment:
Direct database access introduce security risk and by pass validation. We should expose config update via api.

Reply:
In progress to expose APIs.